### PR TITLE
change(doc): add item to release checklist to update dependencies in the README

### PR DESCRIPTION
**Pull request by conradoplg**
## Motivation

Recently we added a `cmake` dependency which we forgot to list in the README, which caused errors for users following the build instructions. In that particular case we were able to change it to a dev-dependency, but the same issue could happen in the future. 

### Designs

Add a item to the checklist to check for new dependencies and list them.

## Solution

Not urgent, anyone can review.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -84,6 +84,9 @@ As we resolve various outstanding known issues and implement new functionality w
 We should check and update if necessary:
 
 - [ ] The "Known Issues" section
+- [ ] The "Build and Run Instructions" section. Check if any new dependencies were introduced and
+      list them if needed; one possible approach is to check for changes in the `Dockerfile`
+      since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.
 
 to ensure that any items that are resolved in the latest release are no longer listed in the README.
```
Comment (upbqdn) [inline .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: Why did you remove the "Known Issues" item? Here's a suggestion that keeps that item, and appends the sentence that was below the list to it.
```suggestion
- [ ] The "Known Issues" section to ensure that any items that are resolved in the latest release are no longer listed in the README.
- [ ] The "Build and Run Instructions" section. Check if any new dependencies were introduced and
      list them if needed; one possible approach is to check for changes in the `Dockerfile`
      since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.
```

Comment (conradoplg) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: I didn't remove it? :sweat_smile: 

But your suggestion is much better, I'll commit it

Comment (upbqdn) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: Oh! :D I'm sorry, I confused the "-" sign indicating a list item with the diff's "removed" indicator. My colorblindness doesn't help with this business.

**Review by upbqdn — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (conradoplg) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: No problem! I figured something like that had happened :laughing:
